# Datasets

The datasets used in the comparisons are detailed below:


## Transcriptomic

### [GSE45291 RA](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE45291)
We analysed a cohort consisting of 493 RA patients and 20 healthy controls. Whole blood samples were subjected to transcriptomic analysis with the Affymetrix HT HG-U133+ platform. The overall design of this experiment was to analyse expression in whole blood from RA and SLE patients, comparing gene expression signatures in patients with SLE, and RA DMARD-IR and RA TNF-IR. For our analysis, in order to select only one disease type, we have eliminated SLE patients.

### [GSE61635](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE61635)
This dataset consists of mRNA levels in blood from SLE patients (79) and healthy controls (30). The samples were sequenced using the Affymetrix Human Genome U133 Plus 2.0 Array platform. The aim of this study was to characterise gene expression profiles in SLE patients with RNP+ autoantibodies versus healthy blood donors, focusing on selected cytokines that may be important in B-cell activation and differentiation, including BAFF, IL-21 and IL-33. They examined a cohort of 79 consecutive patients classified as anti-ribonuclear protein (anti-RNP)+ systemic lupus erythematosus (SLE).

### [GSE11907 T1D GPL97](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE11907)
Transcriptional profiles were generated with Affymetrix U133A and U133B GeneChips (> 44,000 probe sets). 20 type I diabetes were included in this study. In addition, 10 patients were selected as control. 


## Epigenomic

### [GSE42861](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE42861)
Bisulphite converted DNA from the Rheumatoid arthritis patients (cases) and normal controls were hybridized to the Illumina Illumina HumanMethylation450 BeadChip arrays. 689 samples were included in this cohort, of which 335 correspond to healthy samples whereas 354 correspond to RA patients.

### [GSE59250 B cells](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE59250)
This study performed Illumina Methylation450 analysis of several cell lines from peripheral blood in SLE patients. In this study, only B cells were selected. In total, 104 samples were included (56 healthy and 48 SLE).

### [GSE56606 Monocytes](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE56606)
Bisulphite converted DNA from the 56 samples were hybridised to the Illumina Infinium 27k Human Methylation Beadchip v1.2. Methylation profile of monocytes were extracted from 39 healthy patients and 17 T1D patients.

## Metagenomic

### [Gevers et al](https://pubmed.ncbi.nlm.nih.gov/24629344/)



### [Morgan et al](https://pubmed.ncbi.nlm.nih.gov/23013615/)



## Multi-omic experiments

### [GSE117931](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE117931)
This cohort consists in genome-wide DNA methylation and transcription data in peripheral blood mononuclear cells of systemic sclerosis. The cohort involves only 37 samples, of which 19 are healthy individuals and 18 MSc.

### [GSE82221](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE82221)
This cohort has available whole genome transcription and DNA methylation data of peripheral blood mononuclear cells in SLE patients. In total, involves 55 patients (25 healthy and 30 SLE).